src/main/java/core/basesyntax/service/impl/TransactionParserImpl.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.TransactionParser;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TransactionParserImpl implements TransactionParser {
+    private static final String SEPARATOR = ",";
+    private static final int INDEX_CODE_OPERATION = 0;
+    private static final int INDEX_FRUIT_NAME = 1;
+    private static final int INDEX_FRUIT_QUANTITY = 2;
+
+    @Override
+    public List<FruitTransaction> parseData(List<String> data) {
+        try {
+            return data.stream()
+                    .map(this::parseLine)
+                    .collect(Collectors.toList());
+        } catch (NumberFormatException e) {
+            throw new RuntimeException("Invalid value for fruit quantity");
+        }
+    }
+
+    private FruitTransaction parseLine(String line) {
+        String[] elements = line.split(SEPARATOR);

```

Comment:
You can change name of method cause return type of transaction, better is get or parse TransactionFromData or smth like that